slides/intro-extra-recipes.qmd

```diff
@@ -0,0 +1,370 @@
+\ \ [step_normalize]{style="color: #CA225E;"}(all_numeric_predictors())
+:::
+
+<br>
+
+specifying what actions to take by adding `step_*()`s
+
+## What is a recipe?
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_dummy([all_nominal_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_zv([all_predictors()]{style="color: #CA225E;"}) %>%  
+\ \ step_log([distance]{style="color: #CA225E;"}) %>% 
+\ \ step_normalize([all_numeric_predictors()]{style="color: #CA225E;"})
+:::
+<br>
+
+using {tidyselect} and recipes specific selectors to denote affected variables
+
+## Using a recipe
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
+\ \ step_unknown(all_nominal_predictors()) %>%  
+\ \ step_dummy(all_nominal_predictors()) %>%  
+\ \ step_zv(all_predictors()) %>%  
+\ \ step_log(distance) %>% 
+\ \ step_normalize(all_numeric_predictors())
+:::
+
+<br>
+
+recipes are can be used with {workflows} to "combine" it with a model
+
+<br>
+
+:::{style="font-family: 'Source Code Pro', monospace; font-size: 0.8em;"}
+wf_spec <- workflow() %>%  
+\ \ [add_recipe(rec_spec)]{style="color: #CA225E;"} %>%  
+\ \ add_model(linear_reg())
+:::
+
+## recipes are estimated
+
+<br>
+
+Every preprocessing step in a recipe that involved calculations uses the *training* set. For example:
+
+- Levels of a factor
+- Determination of zero-variance
+- Normalization
+- Feature extraction
+
+Once a a recipe is added to a workflow, this occurs when `fit()` is called.
+
+
+## Debugging a recipe
+
+- Typically, you will want to use a workflow to estimate and apply a recipe.
+
+. . .
+
+- If you have an error and need to debug your recipe, the original recipe object (e.g. `rec_spec`) can be estimated manually with a function called `prep()`. It is analogous to `fit()`. See [TMwR section 16.4](https://www.tmwr.org/dimensionality.html#recipe-functions)
+
+. . .
+
+- Another function (`bake()`) is analogous to `predict()`, and gives you the processed data back.
+
+. . .
+
+- The `tidy()` function can be used to get specific results from the recipe.
+
+## Your turn {transition="slide-in"}
+
+![](images/parsnip-flagger.jpg){.absolute bottom="0" left="0" width="150" height="150"}
+
+*Take the recipe, and `prep()`, then `bake()` it to see what the resulting data set looks like*
+
+<br>
+
+*Try removing steps to see how the results changes*
+
+<br>
+
+```{r}
+#| echo: false
+countdown(minutes = 5, id = "recipes-prep")
+```
+
+## printing a recipe
+
+```{r}
+rec_spec <- recipe(tip ~ ., data = taxi_train) %>%  
+  step_unknown(all_nominal_predictors()) %>%  
+  step_dummy(all_nominal_predictors()) %>%  
+  step_zv(all_predictors()) %>%  
+  step_log(distance) %>% 
+  step_normalize(all_numeric_predictors())
+```
```

Comment:
Could this be part of the earlier slides so that we don't need to spend the space on this here?

Reply:
We could, do you have any suggestions on where to put it?

Reply:
I decided to just hide that code chunk. the previous highlighted code sections now do `rec_spec <- ` so it should make sense